tests/unit/drivers/querylang/queryset/test_lookup.py

```diff
@@ -173,3 +175,30 @@ def test_lookup_leaf_None():
     assert leaf.evaluate(mock0)
     mock1 = MockId(4)
     assert not leaf.evaluate(mock1)
+
+def test_docs_filter():
+    s = random_docs(10)
+    ss = QuerySet(s).filter(tags__id__lt=5, tags__id__gt=3)
+    ssr = list(ss)
+    assert len(ssr) == 1
+    for d in ssr:
+        assert (3 < d.tags['id'] < 5)
+
+
+def test_docs_filter_equal():
```

Comment:
There is some redundant code. What about some kind of get_data Funktion?

Reply:
these codes were copy-pasted from another file, I'll refactor this, thanks @florian-hoenicke !

Reply:
created a fixture for this